core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -1231,61 +1394,59 @@ public void removeColumn(CharSequence name) {
         }
 
         if (timestamp == getPartitionLo(maxTimestamp)) {
             LOG.error()
                     .$("cannot remove active partition [path=").$(path)
                     .$(", maxTimestamp=").$ts(maxTimestamp)
-                    .$(']').$();
+                    .I$();
             return false;
         }
 
         if (!txWriter.attachedPartitionsContains(timestamp)) {
-            LOG.error().$("partition is already detached [path=").$(path).$(']').$();
+            LOG.error().$("partition is already detached [path=").$(path).I$();
             return false;
         }
 
-        try {
-            // when we want to delete first partition we must find out
-            // minTimestamp from next partition if it exists or next partition and so on
-            //
-            // when somebody removed data directories manually and then
-            // attempts to tidy up metadata with logical partition delete
-            // we have to uphold the effort and re-compute table size and its minTimestamp from
-            // what remains on disk
-
-            // find out if we are removing min partition
-            long nextMinTimestamp = minTimestamp;
-            if (timestamp == txWriter.getPartitionTimestamp(0)) {
-                nextMinTimestamp = readMinTimestamp(txWriter.getPartitionTimestamp(1));
-            }
-            long partitionNameTxn = txWriter.getPartitionNameTxnByPartitionTimestamp(timestamp);
-            txWriter.beginPartitionSizeUpdate();
-            txWriter.removeAttachedPartitions(timestamp);
-            txWriter.setMinTimestamp(nextMinTimestamp);
-            txWriter.finishPartitionSizeUpdate(nextMinTimestamp, txWriter.getMaxTimestamp());
-            txWriter.bumpTruncateVersion();
-            txWriter.commit(defaultCommitMode, denseSymbolMapWriters);
-
-            // Call O3 methods to remove check TxnScoreboard and remove partition directly
-            o3PartitionRemoveCandidates.clear();
-            o3PartitionRemoveCandidates.add(timestamp, partitionNameTxn);
-            o3ProcessPartitionRemoveCandidates();
+        // when we want to delete first partition we must find out
+        // minTimestamp from next partition if it exists or next partition and so on
+        //
+        // when somebody removed data directories manually and then
+        // attempts to tidy up metadata with logical partition delete
+        // we have to uphold the effort and re-compute table size and its minTimestamp from
+        // what remains on disk
 
-            return true;
-        } finally {
-            path.trimTo(rootLen);
+        // find out if we are removing min partition
+        long nextMinTimestamp = minTimestamp;
+        if (timestamp == txWriter.getPartitionTimestamp(0)) {
+            nextMinTimestamp = readMinTimestamp(txWriter.getPartitionTimestamp(1));
         }
+        long partitionNameTxn = txWriter.getPartitionNameTxnByPartitionTimestamp(timestamp);
```

Comment:
I this we need to remove records from column version file here too, same as detach

```java
columnVersionWriter.detachPartitionColumns(timestamp);
txWriter.setColumnVersion(columnVersionWriter.getVersion());
```

Reply:
done in [47a14c2](https://github.com/questdb/questdb/pull/2280/commits/47a14c2e53d813b5612f72cf047cde353adfd205)